tests/README.md

```diff
@@ -15,6 +15,30 @@ That will run tests on the `/dev/ttyACM0` serial port.  You can also use shortcu
 device names like `a<n>` for `/dev/ttyACM<n>` and `c<n>` for `COM<n>`.  Use
 `./run-tests.py --help` to see all of the device possibilites, and other options.
 
+There are three kinds of tests:
+
+* Tests without a corresponding `.exp` file: these tests are first run under CPython
+  and the output captured, and then run under MicroPython and the output compared to the
+  CPython output.  The test passes if the output matches exactly.  If the output differs
+  then the test fails and the outputs are saved in a `.exp` and a `.out` file
+  respectively.
```

Comment:
This paragraph about tests without exp might make more sense to move down lower now and rewritten slightly to reflect that they're run under cpython only if there's no unittest result returned.

Reply:
OK, I'm moved the unittest paragraph to the top and adjusted the text.